src/lib/services/user-service.ts

```diff
@@ -260,18 +267,35 @@ class UserService {
         email: string,
         autoCreateUser: boolean = false,
     ): Promise<IUser> {
+        return this.loginUserSSO({ email, autoCreate: autoCreateUser });
+    }
+
+    async loginUserSSO({
+        email,
+        name,
+        rootRole,
+        autoCreate = false,
+    }: ILoginUserRequest): Promise<IUser> {
         let user: IUser;
 
         try {
             user = await this.store.getByQuery({ email });
+            // Update user if autCreate is enabled.
+            if (autoCreate) {
+                if (rootRole) {
+                    await this.accessService.setUserRootRole(user.id, rootRole);
```

Comment:
No we cannot change the root role when a user signs in. This can lead to users being downgraded from Admin to Viewer on next sign-in.